luaui/Widgets/unit_smart_area_reclaim.lua

```diff
@@ -169,147 +171,184 @@ local function issue(rList, shift)
 	end
 end
 
+local reclaimOrders = {}
+
+-- we use the previous unit loop iterating on cmds to store reclaim orders
+-- and return cmds for its original usage
+local function storeReclaimOrders(uid)
+		local cmds = GetUnitCommands(uid, maxOrdersCheck)
+
+		reclaimOrders[uid] = {}
+		local reclaimOrdersCount = 0
+
+		for _, order in pairs(cmds) do
+			if order["id"] == RECLAIM then
+				reclaimOrdersCount = reclaimOrdersCount + 1
+				reclaimOrders[uid][reclaimOrdersCount] = order["params"][1]
+			end
+		end
+
+		return cmds
+end
+
+local function checkNoDuplicateOrder(uid, fid)
+	local orderParam = fid+maxUnits
```

Comment:
No idea why reclaim uses fid+maxUnits instead of just fid

Reply:
[Huh](https://github.com/beyond-all-reason/spring/blob/38fe9d715fc44c4b4127bd54ddd9e7b3c3306198/rts/Sim/Units/CommandAI/CommandAI.cpp#L692-L697)

Reply:
because you can reclaim units too, and unitIDs and featureIDs have overlapping number pools (in particular you can have e.g. unitID 123 and featureID 123 at the same time)

Reply:
Ah, that's right xD

Reply:
A bit of a wonky api I think, picking feature or unit as first param would maybe make more sense and be extensible.